packages/next/src/build/swc/index.ts

```diff
@@ -669,10 +671,19 @@ function bindingToApi(binding: any, _wasm: boolean) {
             })
           }
         }
+      } catch (e) {
+        if (e === cancel) return
+        throw e
       } finally {
         binding.rootTaskDispose(task)
       }
     })()
+    iterator.return = async () => {
+      canceled = true
+      if (waiting) waiting.reject(cancel)
+      return { value: undefined, done: true } as IteratorReturnResult<never>
+    }
```

Comment:
The subscription iterator will now break any waiting reads. `AsyncIteartor` will queue the return abrupt completion behind the currently executing read, which means something has to be emitted before the iterator can be cleaned up. This new implementation can break at any point, the read will return, and we'll immediately cleanup.